File: 프로그래머스/unrated/181851. 전국 대회 선발 고사/전국 대회 선발 고사.py
def solution(rank, attendance):
    answer = 0
    arr = [];
    
    for i, r in enumerate(rank):
        if attendance[i]:
            arr.append((r, i))
            
    arr = sorted(arr)[:3]
        
    return arr[0][1] * 10000 + arr[1][1] * 100 + arr[2][1]

File: 프로그래머스/unrated/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;
import java.util.Arrays;

class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        Map<Integer, Integer> m = new HashMap<>();

        for (int i=0; i<rank.length; i++) {
            if (attendance[i]) m.put(rank[i], i);
        }

        List<Integer> x = m.entrySet().stream().map(Map.Entry::getKey).sorted().limit(3).collect(Collectors.toList());

        int result = 10000 * m.get(x.get(0));
        result += 100 * m.get(x.get(1));
        result += m.get(x.get(2));

        return result;
    }
}